target/linux/ath79/dts/ar9330_ziking_cpe46b.dts

```diff
@@ -0,0 +1,111 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "ar9330.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+        model = "ZiKing CPE46B";
+        compatible = "ziking,cpe46b", "qca,ar9330";
+
+        aliases {
+                serial0 = &uart;
+                label-mac-device = &eth0;
+        };
+
+        leds {
+                compatible = "gpio-leds";
+
+                wlan {
+                        label = "green:wlan";
+                        gpios = <&gpio 0 GPIO_ACTIVE_HIGH>;
+                        linux,default-trigger = "phy0tpt";
+                };
+
+                rssi1 {
+                        label = "green:rssi1";
+                        gpios = <&gpio 20 GPIO_ACTIVE_HIGH>;
+                };
+
+                rssi2 {
+                        label = "green:rssi2";
+                        gpios = <&gpio 23 GPIO_ACTIVE_HIGH>;
+                };
+        };
+
+        keys {
+                compatible = "gpio-keys";
+
+                reset {
+                        label = "reset";
+                        linux,code = <KEY_RESTART>;
+                        gpios = <&gpio 12 GPIO_ACTIVE_LOW>;
+                };
+        };
+};
+
+&uart {
+        status = "okay";
+};
+
+&spi {
+        status = "okay";
+
+        num-chipselects = <1>;
+
+        flash@0 {
+                compatible = "jedec,spi-nor";
+                spi-max-frequency = <50000000>;
+                reg = <0>;
+
+                partitions {
+                        compatible = "fixed-partitions";
+                        #address-cells = <1>;
+                        #size-cells = <1>;
+
+                        partition@0 {
+                                label = "u-boot";
+                                reg = <0x000000 0x010000>;
+                                read-only;
+                        };
+
+                        partition@10000 {
+                                label = "u-boot-env";
+                                reg = <0x010000 0x010000>;
+                        };
+
+                        partition@20000 {
+                                compatible = "denx,uimage";
+                                label = "firmware";
+                                reg = <0x020000 0x7d0000>;
+                        };
+
+                        art: partition@7f0000 {
+                                label = "art";
+                                reg = <0x7f0000 0x010000>;
+                                read-only;
+                        };
+                };
+        };
+};
+
+&eth0 {
+        status = "okay";
+
+        mtd-mac-address = <&art 0x0>;
+};
+
+&eth1 {
+        status = "okay";
+
+        mtd-mac-address = <&art 0x0>;
+        mtd-mac-address-increment = <2>;
+};
+
+&wmac {
+        status = "okay";
+
+        mtd-cal-data = <&art 0x1000>;
+        mtd-mac-address = <&art 0x1002>;
```

Comment:
mtd-mac-address can be dropped, it's read automatically from the cal-data ...

Reply:
Ok thanks